raijin_scripts/execute_s2_to_s3_rolling/s2_to_s3_rolling.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3
+import datetime
+import logging
+import os
+import subprocess
+import sys
+
+import boto3
+import botocore
+import yaml
+from odc.index import odc_uuid
+
+handler = logging.StreamHandler()
+LOG = logging.getLogger("s3_to_s3_rolling")
+LOG.setLevel(logging.DEBUG)
+LOG.addHandler(handler)
+
+NCI_DIR = '/g/data/if87/datacube/002/S2_MSI_ARD/packaged'
+S3_PATH = 'L2/sentinel-2-nbar/S2MSIARD_NBAR'
+
+
+def find_granules(num_days, end_date, root_path=NCI_DIR):
+    # Find the dates between the input date and today, inclusive, formatted like the directories
+    dates = [(end_date - datetime.timedelta(days=x)).strftime("%Y-%m-%d") for x in range(num_days + 1)]
+
+    # The list of folders will be returned and will contain all the granules available for
+    # the date range specified above. Format is yyyy-mm-dd/granule
+    list_of_granules = []
+
+    for date in dates:
+        dir_for_date = os.path.join(root_path, date)
+        if os.path.exists(dir_for_date):
+            granules = [date + "/" + name for name in os.listdir(dir_for_date)]
+            list_of_granules += granules
+
+    return(list_of_granules)
+
+
+def check_granule_exists(s3_bucket, s3_metadata_path):
+    s3 = boto3.resource('s3')
```

Comment:
This probably should be shared, I think it's slower to recreated it every time (but I can't remember by how much).